src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/wrapper/Functions.java

```diff
@@ -964,16 +1034,65 @@ private static void addMGF(long id, String name) {
+        addMech(CKM_SHA512_T_KEY_GEN,           "CKM_SHA512_T_KEY_GEN");
+        addMech(CKM_NULL,                       "CKM_NULL");
+        addMech(CKM_BLAKE2B_160,                "CKM_BLAKE2B_160");
+        addMech(CKM_BLAKE2B_160_HMAC,           "CKM_BLAKE2B_160_HMAC");
+        addMech(CKM_BLAKE2B_160_HMAC_GENERAL,   "CKM_BLAKE2B_160_HMAC_GENERAL");
+        addMech(CKM_BLAKE2B_160_KEY_DERIVE,     "CKM_BLAKE2B_160_KEY_DERIVE");
+        addMech(CKM_BLAKE2B_160_KEY_GEN,        "CKM_BLAKE2B_160_KEY_GEN");
+        addMech(CKM_BLAKE2B_256,                "CKM_BLAKE2B_256");
+        addMech(CKM_BLAKE2B_256_HMAC,           "CKM_BLAKE2B_256_HMAC");
+        addMech(CKM_BLAKE2B_256_HMAC_GENERAL,   "CKM_BLAKE2B_256_HMAC_GENERAL");
+        addMech(CKM_BLAKE2B_256_KEY_DERIVE,     "CKM_BLAKE2B_256_KEY_DERIVE");
+        addMech(CKM_BLAKE2B_256_KEY_GEN,        "CKM_BLAKE2B_256_KEY_GEN");
+        addMech(CKM_BLAKE2B_384,                "CKM_BLAKE2B_384");
+        addMech(CKM_BLAKE2B_384_HMAC,           "CKM_BLAKE2B_384_HMAC");
+        addMech(CKM_BLAKE2B_384_HMAC_GENERAL,   "CKM_BLAKE2B_384_HMAC_GENERAL");
+        addMech(CKM_BLAKE2B_384_KEY_DERIVE,     "CKM_BLAKE2B_384_KEY_DERIVE");
+        addMech(CKM_BLAKE2B_384_KEY_GEN,        "CKM_BLAKE2B_384_KEY_GEN");
+        addMech(CKM_BLAKE2B_512,                "CKM_BLAKE2B_512");
+        addMech(CKM_BLAKE2B_512_HMAC,           "CKM_BLAKE2B_512_HMAC");
+        addMech(CKM_BLAKE2B_512_HMAC_GENERAL,   "CKM_BLAKE2B_512_HMAC_GENERAL");
+        addMech(CKM_BLAKE2B_512_KEY_DERIVE,     "CKM_BLAKE2B_512_KEY_DERIVE");
+        addMech(CKM_BLAKE2B_512_KEY_GEN,        "CKM_BLAKE2B_512_KEY_GEN");
+        addMech(CKM_SALSA20,                    "CKM_SALSA20");
+        addMech(CKM_CHACHA20_POLY1305,          "CKM_CHACHA20_POLY1305");
+        addMech(CKM_SALSA20_POLY1305,           "CKM_SALSA20_POLY1305");
+        addMech(CKM_X3DH_INITIALIZE,            "CKM_X3DH_INITIALIZE");
+        addMech(CKM_X3DH_RESPOND,               "CKM_X3DH_RESPOND");
+        addMech(CKM_X2RATCHET_INITIALIZE,       "CKM_X2RATCHET_INITIALIZE");
+        addMech(CKM_X2RATCHET_RESPOND,          "CKM_X2RATCHET_RESPOND");
+        addMech(CKM_X2RATCHET_ENCRYPT,          "CKM_X2RATCHET_ENCRYPT");
+        addMech(CKM_X2RATCHET_DECRYPT,          "CKM_X2RATCHET_DECRYPT");
+        addMech(CKM_XEDDSA,                     "CKM_XEDDSA");
+        addMech(CKM_HKDF_DERIVE,                "CKM_HKDF_DERIVE");
+        addMech(CKM_HKDF_DATA,                  "CKM_HKDF_DATA");
+        addMech(CKM_HKDF_KEY_GEN,               "CKM_HKDF_KEY_GEN");
+        addMech(CKM_SALSA20_KEY_GEN,            "CKM_SALSA20_KEY_GEN");
+        addMech(CKM_SP800_108_COUNTER_KDF,      "CKM_SP800_108_COUNTER_KDF");
+        addMech(CKM_SP800_108_FEEDBACK_KDF,     "CKM_SP800_108_FEEDBACK_KDF");
+        addMech(CKM_SP800_108_DOUBLE_PIPELINE_KDF,
+                                         "CKM_SP800_108_DOUBLE_PIPELINE_KDF");
```

Comment:
same comment as above.

Reply:
Well, per the ordering in PKCS11Constants, these three lines are at the right place. Note that the ordering of CKM_ECDSA_SHA3_224 to CKM_EDDSA in pkcs11t.h is different from PKCS11Constants class, so I will add the comment about the general ordering following PKCS11Constants class and keep them here.

Reply:
Sounds good that a comment will be added.